portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,17 +107,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String argURL = request.getParameter("url-input");
     String argComment = request.getParameter("comment-input");
+
+    if(
+        !(
+            ( Pattern.matches("^http:\\/\\/(.+)\\.sinaimg\\.cn\\/large\\/(.+)\\.(jpg|png|gif)$",argURL) )
```

Comment:
Don't need to escape slash "/" in a regular expression.

I have tried to upload a meme using one of the provided sina URL, but failed. Maybe the escape is the root cause. Try fixing the pattern?

![image](https://user-images.githubusercontent.com/635513/87412813-0d12fc80-c5fc-11ea-947a-e62ff6cf45bc.png)

![image](https://user-images.githubusercontent.com/635513/87412967-3b90d780-c5fc-11ea-97b4-f592937e84c9.png)